projects/ngneat/transloco/src/lib/transloco.service.ts

```diff
@@ -102,11 +118,14 @@ export class TranslocoService implements OnDestroy {
   load(lang: string, options?: { fallbackLangs: string[] | null }): Observable<Translation> {
     if (this.cache.has(lang) === false) {
       const mergedOptions = { ...{ fallbackLangs: null }, ...(options || {}) };
-
       const load$ = from(this.loader.getTranslation(lang)).pipe(
         retry(this.mergedConfig.failedRetries),
         catchError(() => this.handleFailure(lang, mergedOptions)),
-        tap(translation => this.handleSuccess(lang, translation)),
+        withLatestFrom(this.loadFallbackLang(lang)),
+        map(([translations, fallbackTranslations]) =>
+          fallbackTranslations ? { ...translations, ...fallbackTranslations } : translations
```

Comment:
It should be { ...fallbackTranslations, ...translations }

Reply:
But it still doesn't accurate. We need to support empty values:

// es.json
```json
{
  someKey: ""
}
```

// en.json
```json
{
  someKey: "There is a value"
}
```

The merge should return:
```json
{
  someKey: "There is a value"
}
```

Reply:
You can resolve it by skip the merging and use it in the missing handler. V2 will support it.

Reply:
I'm agree the fallback should handle empty values but in this case we also need to rename the `useFallbackForMissingKey` option by something like `useFallbackForMissingTranslation` which is more accurate IMO. Also I'm not sure to understand what you mean by skip the merging and use it in the missing handler.

Reply:
Right, it will be easier and cleaner to handle missing key/value in v2. I will do this soon. Do you have an idea of when v2 would be pushed?

Reply:
https://github.com/ngneat/transloco/tree/v2

You can work directly on this branch.